### DISCUSSION TITLE
Subtitle timings are not precise

### DISCUSSION THREAD

**Discussion by fadyscube** [openai/whisper#1147] [Q&A] [answered]
When whisper generates the srt file, the subtitles are not synchronized with the speech, like when there are silents in the speech, the subs timings include the silence, I wanted to know if there is a way to have the subs timing synchronized with the speech.

**Comment by ryanheise** [comment, marked as answer]
You can try #1114 in conjunction with the `--word_timestamps True` option, which improves timestamp accuracy when there is silence between sentences.

Note however that `--word_timestamps True` currently also generates a different type of srt file where individual words are highlighted. In the future, word highlighting should probably be an option.

**Reply by fadyscube** [reply]
Is there a way to do this with the API

**Comment by ioudove** [comment]
> 当耳语生成srt文件时，字幕与语音不同步，就像语音中无声时，子时序包括静音，我想知道是否有办法使子时序与语音同步。

Hi, I am also plagued by this problem, have you found a solution to this problem?

**Comment by Edweis** [comment]
A 3y old thread but I found https://github.com/m-bain/whisperx working a lot better for me.
